plugin.js

```diff
@@ -66,10 +57,27 @@ function onReqHandlerWrapper (fastify, hook) {
       if (cookieHeader) {
         fastifyReq.cookies = fastify.parseCookie(cookieHeader)
       }
+      fastifyRes[kReplySetCookies] = new Map()
       done()
     }
 }
 
+function onSendHandlerWrapper (fastify) {
+  return function fastifyCookieHandler (fastifyReq, fastifyRes, payload, done) {
+    if (fastifyRes[kReplySetCookies].size) {
+      const setCookie = []
+
+      for (const [, c] of fastifyRes[kReplySetCookies]) {
+        setCookie.push(fastify.serializeCookie(c.name, c.value, c.opts))
+      }
+
+      fastifyRes.header('Set-Cookie', setCookie)
+    }
+
+    done()
+  }
+}
+
```

Comment:
Why do we need to wrap it? 
If we directly use serializeCookie then we dont need to wrap?
Also we know the size of the array beforehand, so maybe better to pre initialize it .
Maybe needs some benching?

```suggestion
function fastifyCookieOnSendHandler (fastifyReq, fastifyRes, payload, done) {
  if (fastifyRes[kReplySetCookies].size) {
    const setCookie = Array(fastifyRes[kReplySetCookies].size)

    let i = 0
    for (const [, c] of fastifyRes[kReplySetCookies]) {
      setCookie[i++] = serializeCookie(c.name, c.value, c.opts)
    }

    fastifyRes.header('Set-Cookie', setCookie)
  }
  done()
}

```

Reply:
> Also we know the size of the array beforehand, so maybe better to pre initialize it.

I don't know if ECMAScript has preinitialization of arrays. I can do it if V8 optimizes it though

> Why do we need to wrap it?

I agree that it's not necessary, just followed the same API pattern where we pass fastify to use the decorated method instead of the *real* one. If I remove it and directly use the functions, it makes sense to also remove it in the other wrapper

Reply:
If you know the size of an array, it is faster to set the size and directly set the values than just initializing the array. This is basically in all languages like this, because if an array a has not known size than the runtime has to allocate memory for the array dynamically. Like if you have an empty array and add an element the runtime has to increase the size of the array by one. memory allocation/dynamic resizing is usually in any language/runtime perf bottleneck. If you know before hand that you have x elements, the runtime can allocate the size of the array beforehand avoiding the dynamic resizing.